drivers/vdpa/virtio/virtio_vdpa.c

```diff
@@ -1336,6 +1473,46 @@ virtio_vdpa_dev_config(int vid)
 	DRV_LOG(INFO, "%s vid %d was configured, took %lu us.", vdev->device->name,
 			vid, (t_end - t_start) * 1000000 / rte_get_tsc_hz());
 
+	if (!priv->ctx_stored) {
+		if (!priv->restored_from_ctx) {
+			/* If we restored from cached_ctx in probe, devargs and fds should be the same,
+			 * so don't store them again
+			 */
+			strcpy(vf_dev.vf_name.dev_bdf, priv->vf_name.dev_bdf);
+			rte_uuid_unparse(priv->vm_uuid, vf_dev.vm_uuid, RTE_UUID_STRLEN);
+
+			ret = rte_vhost_get_ifname(vid, vf_dev.vhost_sock_addr, VDPA_MAX_SOCK_LEN);
+			if (ret) {
+				DRV_LOG(ERR, "Failed to get vhost sock addr (vid %d)", vid);
+				return 0;
+			}
+
+			ret = virtio_ha_vf_devargs_fds_store(&vf_dev, &priv->pf_name, priv->vfio_container_fd,
+				priv->vfio_group_fd, priv->vfio_dev_fd);
+			if (ret) {
+				DRV_LOG(ERR, "Failed to store vf devargs and vfio fds (vid %d)", vid);
+				return 0;
+			}
+		}
+
+		vhost_sock_fd = rte_vhost_get_conn_fd(vid);
+		if (vhost_sock_fd < 0) {
+			DRV_LOG(ERR, "Failed to get vhost sock fd (vid %d)", vid);
+			return 0;
+		}
+
+		ret = virtio_ha_vf_vhost_fd_store(&priv->vf_name, &priv->pf_name, vhost_sock_fd);
+		if (ret) {
+			DRV_LOG(ERR, "Failed to store vhost fd (vid %d)", vid);
+			return 0;			
+		}
+
+		priv->ctx_stored = true;
```

Comment:
priv->ctx_stored should be flase after dev_close

Reply:
done